src/python/pants/goal/context.py

```diff
@@ -144,6 +142,11 @@ def scm(self):
     """
     return self._scm
 
+  @property
+  def scheduler(self):
```

Comment:
@ity FYI, this may be useful :)

Reply:
If possible, it would be good not to expose this as a public-looking API, and continue to access the private field (or to expose a method that closes over the entire "get the graph len and record it in the runtracker" operation). The goal of #4769 is to give Tasks the ability to request that things run in the scheduler in a declarative way. Having imperative access is supposed to just be a stopgap.

Reply:
sgtm - fixed.